src/Query.php

```diff
@@ -454,7 +454,18 @@ public function assembleSelect()
         foreach ($resolved as $target) {
             $targetColumns = $resolved[$target]->getArrayCopy();
             if (isset($omitted[$target])) {
-                $targetColumns = array_diff($targetColumns, $omitted[$target]->getArrayCopy());
+                $toExclude = $omitted[$target]->getArrayCopy();
+                $targetColumns = array_filter($targetColumns, function ($value, $column) use ($toExclude) {
```

Comment:
These names confuse me a little. Please use `$column` and `$alias` instead, respectively.